packages/compass-signtool/src/signing-clients/local-signing-client.ts

```diff
@@ -0,0 +1,56 @@
+import path from 'path';
+import { exec } from 'child_process';
+import { promisify } from 'util';
+import { debug } from '../utils';
+import type { SigningClient, SigningClientOptions } from '.';
+
+const execAsync = promisify(exec);
+
+export class LocalSigningClient implements SigningClient {
+  constructor(private options: SigningClientOptions) {}
+
+  private async init() {
+    const remoteScript = `${this.options.rootDir}/garasign.sh`;
+    await execAsync(`mkdir -p ${this.options.rootDir}`);
+    await this.copyFile(this.options.signingScript, remoteScript);
+    await execAsync(`chmod +x ${remoteScript}`);
+  }
+
+  private async copyFile(from: string, to: string): Promise<void> {
+    await execAsync(`cp ${from} ${to}`);
+  }
+
+  async sign(file: string): Promise<void> {
+    debug('Signing file', file);
+
+    const remotePath = path.join(this.options.rootDir, path.basename(file));
+
+    try {
+      await this.init();
+
+      await this.copyFile(file, remotePath);
+      debug(`LocalSigningClient: Copied file ${file} to ${remotePath}`);
+
+      await execAsync(
+        `cd ${this.options.rootDir} && ./garasign.sh ${path.basename(file)}`,
```

Comment:
`child_process.execFile` takes a `cwd` option so you don't need to manually `cd` into a specific directory first